src/wp-includes/user.php

```diff
@@ -1849,9 +1849,18 @@ function update_user_caches( $user ) {
 	}
 
 	wp_cache_add( $user->ID, $user, 'users' );
-	wp_cache_add( $user->user_login, $user->ID, 'userlogins' );
-	wp_cache_add( $user->user_email, $user->ID, 'useremail' );
-	wp_cache_add( $user->user_nicename, $user->ID, 'userslugs' );
+
+	if ( ! empty( $user->user_login ) ) {
```

Comment:
What if the string is "false" or "0"?

Reply:
Resolved in acc52430e43916f7e116d77bc3098029927cc5b5.